### PR TITLE
Fix empty prop of tag and edge

### PR DESCRIPTION
**Pull request by laura-ding**
Delete [] of SRC_REF and DST_REF
Fix insert vertex and edge when tag or edge has the empty prop

### CODE REVIEW

File: src/graph/test/DataTest.cpp [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "CREATE TAG company(name string)";
+        auto code = client_->execute(cmd, resp);
+        if (cpp2::ErrorCode::SUCCEEDED != code) {
+            return TestError() << "Do cmd:" << cmd << " failed";
+        }
+    }
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "CREATE EDGE work()";
```
Comment (whitewum) [inline src/graph/test/DataTest.cpp]: add some cases:
1. create edge e1(); create edge e2(); 
2. create tag t1(); create e1();
3. a very very long sentence /* or repeatly add 1000 times  */

File: src/graph/test/DataTest.cpp [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "INSERT EDGE work() VALUES 10001->20001:()";
+        auto code = client_->execute(cmd, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+    }
+    // Get result
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "GO FROM 10001 OVER work yield $$company.name";
```
Comment (whitewum) [inline src/graph/test/DataTest.cpp]: add some case:
try to get a property which does not exist;

File: src/graph/test/SchemaTest.cpp [hunk 3]
```diff
@@ -184,21 +203,28 @@ TEST_F(SchemaTest, metaCommunication) {
         };
         ASSERT_TRUE(verifyResult(resp, expected));
     }
+    // Test create edge without prop
     {
-        // Test tag not exist
         cpp2::ExecutionResponse resp;
-        std::string query = "DESCRIBE TAG not_exist";
+        std::string query = "CREATE EDGE edge1()";
```
Comment (whitewum) [inline src/graph/test/SchemaTest.cpp]: add case:
alter edge1 without property to add a property.

Comment (laura-ding) [reply src/graph/test/SchemaTest.cpp]: good

File: src/graph/test/DataTest.cpp (hunk 2 above)
Comment (laura-ding) [reply src/graph/test/DataTest.cpp]: good

**Comment by laura-ding** [comment]
close #453

File: src/graph/test/DataTest.cpp (hunk 1 above)
Comment (steppenwolfyuetong) [reply src/graph/test/DataTest.cpp]: @whitewum  👍
> 3\. a very very long sentence /* or repeatly add 1000 times  */

You inspired me, I have a few small questions:

1) Is there a limit to the length of the SQL statement?
2) Is there a limit on the number of tag/edge columns?
3) Is there a limit to the length of each record when inserting vertex/edge?

File: src/graph/test/DataTest.cpp [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+std::unique_ptr<GraphClient>         DataTest::client_{nullptr};
+
+AssertionResult DataTest::prepareSchema() {
+    {
+        cpp2::ExecutionResponse resp;
+        std::string host = folly::stringPrintf("127.0.0.1:%u", gEnv->storageServerPort());
+        std::string cmd = "ADD HOSTS " + host;
+        auto code = client_->execute(cmd, resp);
+        if (cpp2::ErrorCode::SUCCEEDED != code) {
+            return TestError() << "Do cmd:" << cmd << " failed";
```
Comment (steppenwolfyuetong) [inline src/graph/test/DataTest.cpp]: What about ”Execute cmd“ ？

File: src/graph/test/DataTest.cpp (hunk 1 above)
Comment (whitewum) [reply src/graph/test/DataTest.cpp]: Good! Add tests to cover this limits (and will not be changed by others in the future.)

**Comment by laura-ding** [comment]
close #436

**Comment by whitewum** [comment]
any updates for this PR? @laura-ding Ready for review?

**Comment by laura-ding** [comment]
@whitewum  No anything to update, but should resolve conflicts.

File: src/graph/test/OrderByTest.cpp [hunk 5]
```diff
@@ -40,8 +40,8 @@ TEST_F(OrderByTest, SyntaxError) {
         cpp2::ExecutionResponse resp;
         auto &player = players_["Boris Diaw"];
         auto *fmt = "GO FROM %ld OVER serve YIELD "
-                    "$^[player].name as name, serve.start_year as start, $$[team].name"
-                    "| ORDER BY $-.$$[team].name";
+                    "$^player.name as name, serve.start_year as start, $$[team].name"
```
Comment (whitewum) [inline src/graph/test/OrderByTest.cpp]: $$[team].name
or $$team.name

File: src/graph/test/OrderByTest.cpp [hunk 6]
```diff
@@ -61,7 +61,7 @@ TEST_F(OrderByTest, NoInput) {
         cpp2::ExecutionResponse resp;
         auto &player = players_["Nobody"];
         auto *fmt = "GO FROM %ld OVER serve YIELD "
-                    "$^[player].name as name, serve.start_year as start, $$[team].name as name"
+                    "$^player.name as name, serve.start_year as start, $$team.name as name"
```
Comment (whitewum) [inline src/graph/test/OrderByTest.cpp]: great!

File: src/graph/test/SchemaTest.cpp [hunk 7]
```diff
@@ -111,8 +111,36 @@ TEST_F(SchemaTest, metaCommunication) {
...
+        cpp2::ExecutionResponse resp;
+        std::string query = "CREATE TAG tag1()";
+        auto code = client->execute(query, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+    }
+    {
+        cpp2::ExecutionResponse resp;
+        std::string query = "DESCRIBE TAG tag1";
+        auto code = client->execute(query, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
```
Comment (whitewum) [inline src/graph/test/SchemaTest.cpp]: I think we should check the detailed results of "Describe TAG tag1", i.e., the props is empty

File: src/graph/test/SchemaTest.cpp [hunk 8]
```diff
@@ -171,62 +200,88 @@ TEST_F(SchemaTest, metaCommunication) {
...
         ASSERT_TRUE(verifyResult(resp, expected));
     }
+    // Test existent tag
     {
         cpp2::ExecutionResponse resp;
-        std::string query = "CREATE TAG account(id int, balance double)";
+        std::string query = "CREATE TAG person(id int, balance double)";
         auto code = client->execute(query, resp);
-        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+        ASSERT_NE(cpp2::ErrorCode::SUCCEEDED, code);
```
Comment (whitewum) [inline src/graph/test/SchemaTest.cpp]: Do we have other return code?
 such as ALREAD_EXISTED.
!= SUCCEED is not precise.

File: src/graph/test/SchemaTest.cpp [hunk 9]
```diff
@@ -313,7 +381,42 @@ TEST_F(SchemaTest, metaCommunication) {
         };
         ASSERT_TRUE(verifyResult(resp, expected));
     }
-
+    // Test multi sentences
+    {
+        cpp2::ExecutionResponse resp;
+        std::string query;
+        for (auto i = 0u; i < 1000; i++) {
```
Comment (whitewum) [inline src/graph/test/SchemaTest.cpp]: Great!

File: src/graph/test/SchemaTest.cpp [hunk 10]
```diff
@@ -313,7 +381,42 @@ TEST_F(SchemaTest, metaCommunication) {
...
+            query += "CREATE TAG tag10" + std::to_string(i) + "(name string);";
+        }
+        auto code = client->execute(query, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+    }
+    {
```
Comment (whitewum) [inline src/graph/test/SchemaTest.cpp]: let's suppose tag100~tag101000 have been created.
then let's
1. drop tag10100~tag10300. 
2. alter tag10400~tag10500
let's check the results.
1. -> not existed
2. -> changed
3. -> others, remain unchanged.

File: src/graph/test/OrderByTest.cpp [hunk 11]
```diff
@@ -61,7 +61,7 @@ TEST_F(OrderByTest, NoInput) {
         cpp2::ExecutionResponse resp;
         auto &player = players_["Nobody"];
         auto *fmt = "GO FROM %ld OVER serve YIELD "
-                    "$^player.name as name, serve.start_year as start, $$team.name as name"
+                    "$^.player.name as name, serve.start_year as start, $$.team.name as name"
```
Comment (whitewum) [inline src/graph/test/OrderByTest.cpp]: upper AS

File: src/graph/test/SchemaTest.cpp (hunk 7 above)
Comment (whitewum) [reply src/graph/test/SchemaTest.cpp]: great

**Pull request merged**